Assets/Scripts/Obstacles/ExplodingBarrel.cs

```diff
@@ -0,0 +1,25 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class ExplodingBarrel : MonoBehaviour
+{
+
+    [SerializeField]
+    private MeshRenderer barrelMesh;
+    
+    void Start()
+    {
+        GetComponent<HealthController>().onDeath += Explode;
+    }
+
+    private void Explode(HealthController controller, float damage, DamageInfo info)
+    {
+        barrelMesh.enabled = false;
+        GetComponentInChildren<CapsuleCollider>().enabled = false;
+        GetComponent<ExplotionController>().Explode();
+        
```

Comment:
Perhaps this object should destroy itself after its explosion has finished? (this isn't crucial)

Reply:
Fixed destroys after 4 seconds so the vfx plays